server/clientFactory.js

```diff
@@ -13,12 +21,27 @@ const Config = require('bcfg');
  * a Node, Wallet, and Multisig Wallet clients as available
  */
 function clientFactory(config) {
+  let Network, NodeClient, WalletClient;
   assert(
     config instanceof Config,
     'Must pass instance of Config class to client composer'
   );
 
-  // use network fallbacks
+  // bitcoin, bitcoincash, handshake
+  const chain = config.str('chain', 'bitcoin');
+
+  // set tools based on chain
+  if (chain === 'handshake') {
+    Network = HSNetwork;
+    NodeClient = HSNodeClient;
+    WalletClient = HSWalletClient;
+  } else {
+    // bitcoin settings as fallback
```

Comment:
`// bitcoin/bitcoin cash settings as fallback`

Reply:
This may not be true for long actually as the bcash client might have to start deviating unfortunately :-/. And actually, I haven't tested yet, but may already have to deviate for cash by adding custom port.

Reply:
👍

Reply:
Maybe when it does deviate and we add support here, we can refactor the logic around selecting the correct clients into a module, maybe we could put it into `bpanel-utils` so that the front end and the back end can use the same module?

Reply:
That's a nice idea. Just a simple module that takes in `chain` argument and returns a map of clients? Not sure how to generalize it though to support things like `Network` as well any of the other diverged primitives though. I guess you could have a function that takes a `chain` and a module name and then it tries to extract and return that module from the matching chain's library?

Reply:
We could keep a hardcoded map of maps? Since its only 3 clients, I don't think that hardcoding will run into scaling issues too quickly, but still not perfectly ideal

Reply:
Each submap would contain the modules specific to that project

Reply:
We could write a function that takes the module and builds the object and invoke it 3 times, assuming the shapes of the modules are similar